src/csharp/Microsoft.Spark/Sql/Types/SimpleTypes.cs

```diff
@@ -81,9 +81,9 @@ public sealed class DateType : AtomicType
         /// </summary>
         internal override object FromInternal(object obj)
         {
-            if (obj == null)
+            if (obj == null || obj.GetType() == typeof(Date))
```

Comment:
This seems fishy.  You shouldn't be passing in an already converted type to this method.

Reply:
Changed.